### DISCUSSION TITLE
For those people want hevc preview with hardware acceleration on linux+wayland with mesa driver (amd/intel)

### DISCUSSION THREAD

**Discussion by GRFTSOL** [mifi/lossless-cut#2261] [General]
TLDR: 
Edit the `.desktop` file, change `Exec` to below: 
```Exec=losslesscut --enable-features=AcceleratedVideoDecodeLinuxGL,VaapiVideoDecodeLinuxGL,VaapiVideoDecoder,VaapiIgnoreDriverChecks,UseChromeOSDirectVideoDecoder,Vulkan,DefaultANGLEVulkan,VulkanFromANGLE```

For those not WOOB linux distro like archlinux, before doing this, you need: 
1. mesa driver, eg: `mesa` package on archlinux
2. vulkan driver, eg: `vulkan-radeon` package on archlinux, which is recommended on amdgpu currently.
3. ensure you don't have any other vulkan driver like `amdvlk` on archlinux, in case wrong driver choice by env
> - If other package (eg: `steam`) need 32bit vulkan driver, you need to install `lib32-vulkan-radeon`, they are all belongs to RADV
> - use `ls /usr/share/vulkan/icd.d/` to confirm you have correct vulkan driver, for me on amdgpu, it should be `radeon_icd.i686.json` and `radeon_icd.x86_64.json`
4. Edit the `.desktop` file like above mentioned.

Reference: 
https://wiki.archlinux.org/title/Chromium#Hardware_video_acceleration

Note
- If you really need an extra vulkan driver, please refer this: https://wiki.archlinux.org/title/Vulkan#Switching_between_AMD_drivers
- I don't have Nvidia gpu, but it should be similar, the key is arguments, nvidia driver should already contain vulkan support.
- Please ensure it contain at most one argument starting with `--enable-features`
- For me, latest release of `lossless-cut` is already wayland support, but if you are using old version, maybe you need to refer this: https://wiki.archlinux.org/title/Chromium#Native_Wayland_support to add arguments with running on wayland

How to tell if hwaccel is actually enabled:

![图片](https://github.com/user-attachments/assets/42aaecd0-2597-4c15-a55e-d07966e75421)

1. Dev tools -> hamburger menu -> more tools -> media, check the decoder name, like the picture above.
2. Using `nvtop`, check the `DEC` engine usage, if you don't see `DEC`, means hwaccel not enabled.
3. Of course, see the video preview.

**Comment by mifi** [comment]
thanks i will add it to the faq

**Comment by joshrivers** [comment]
Thanks for the writeup, GRFTSOL. 

I could not get it to work on my NVIDIA system. I went through the entire Arch HW Acceleration guide and verified that the drivers are all working, but no combination of environment variables or chromium flags got me accelerated playback showing up in the dev tools. My file works cleanly on my Windows boot on the same system. There's probably some additional configuration that I'm missing locally, but I'm not sure where to dig next.

**Reply by GRFTSOL** [reply]
Hi Josh, 

For nvidia, I think it should work out of the box if install the needed package, you may turn to [archlinux's bbs](https://bbs.archlinux.org/) for some search, or see archwiki's `nvidia` part, your problem is not belong to this repo, and since I don't have nvidia cards, I may not help much:)

You can also follow people's step on archlinux bbs to check your setup, that's what I always do.
